File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/list/DebtsMVP.java
package com.androidcollider.easyfin.debts.list;

import com.androidcollider.easyfin.common.models.Debt;

import java.util.List;

import io.reactivex.Flowable;

/**
 * @author Ihor Bilous
 */

public interface DebtsMVP {

    interface Model {

        Flowable<List<DebtViewModel>> getDebtList();

        Flowable<Debt> getDebtById(int id);

        Flowable<Boolean> deleteDebtById(int id);
    }

    interface View {

        void setDebtList(List<DebtViewModel> debtList);

        void goToEditDebt(Debt debt, int mode);

        void goToPayDebt(Debt debt, int mode);

        void deleteDebt();
    }

    interface Presenter {

        void setView(DebtsMVP.View view);

        void loadData();

        void getDebtById(int id, int mode, int actionType);

        void deleteDebtById(int id);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/list/DebtsMVP.kt
package com.androidcollider.easyfin.debts.list

import com.androidcollider.easyfin.common.models.Debt
import io.reactivex.rxjava3.core.Single

/**
 * @author Ihor Bilous
 */
interface DebtsMVP {
    interface Model {
        val debtList: Single<List<DebtViewModel>>
        fun getDebtById(id: Int): Single<Debt>
        fun deleteDebtById(id: Int): Single<Boolean>
    }

    interface View {
        fun setDebtList(debtList: List<DebtViewModel>)
        fun goToEditDebt(debt: Debt?, mode: Int)
        fun goToPayDebt(debt: Debt?, mode: Int)
        fun deleteDebt()
    }

    interface Presenter {
        fun setView(view: View?)
        fun loadData()
        fun getDebtById(id: Int, mode: Int, actionType: Int)
        fun deleteDebtById(id: Int)
    }
}